internal/trie/node/stats.go

```diff
@@ -0,0 +1,30 @@
+package node
+
+// Stats contains statistical information for a node.
+type Stats struct {
```

Comment:
Do we really need this `Stats` type? You are adding receiver methods to `Branch` to make this work already, so why not just add a `Descendants uint32` attribute to `Branch`?

Reply:
nit: I would've introduced `Stats` when you have more than just `Descendants`.

Reply:
Agreed, I removed it, only took me a minute 😉